vm/mx.vm/launcher_template_modules.sh

```diff
@@ -0,0 +1,148 @@
+    absolute_module_path+=("${location}/${e}")
+done
+
+IFS=: read -ra relative_upgrade_module_path <<< "<upgrade_module_path>"
+absolute_upgrade_module_path=()
+for e in "${relative_upgrade_module_path[@]}"; do
+    absolute_upgrade_module_path+=("${location}/${e}")
+done
+
+echo "upgrade_module_path=${relative_upgrade_module_path[@]} vs <upgrade_module_path>"
+
+jvm_args=("-Dorg.graalvm.launcher.shell=true")
+launcher_args=()
+
+# Unfortunately, parsing of `--jvm.*` arguments has to be done blind:
+# Maybe some of those arguments where not really intended for the launcher but were application arguments
+
+for o in "$@"; do
+    if [[ "$o" == --jvm.* ]]; then
+        >&2 echo "'--jvm.*' options are deprecated, use '--vm.*' instead."
+        jvm_arg="${o#--jvm.}"
+        if [[ "$jvm_arg" == "cp" ]]; then
+            >&2 echo "'--jvm.cp' argument must be of the form '--jvm.cp=<classpath>', not two separate arguments"
+            exit 1
+        fi
+        if [[ "$jvm_arg" == "classpath" ]]; then
+            >&2 echo "'--jvm.classpath' argument must be of the form '--jvm.classpath=<classpath>', not two separate arguments"
+            exit 1
+        fi
+        if [[ "$jvm_arg" == "cp="* ]]; then
+            custom_cp=${jvm_arg#cp=}
+        elif [[ "$jvm_arg" == "classpath="* ]]; then
+            custom_cp=${jvm_arg#classpath=}
+        fi
+        if [[ -z "${custom_cp+x}" ]]; then
+            jvm_args+=("-${jvm_arg}")
+        else
+            IFS=: read -ra custom_cp_a <<< "${custom_cp}"
+            for e in "${custom_cp_a[@]}"; do
+                absolute_cp+=("${e}")
+            done
+        fi
+    elif [[ "$o" == --vm.* ]]; then
+        vm_arg="${o#--vm.}"
+        if [[ "$vm_arg" == "cp" ]]; then
+            >&2 echo "'--vm.cp' argument must be of the form '--vm.cp=<classpath>', not two separate arguments"
+            exit 1
+        fi
+        if [[ "$vm_arg" == "classpath" ]]; then
+            >&2 echo "'--vm.classpath' argument must be of the form '--vm.classpath=<classpath>', not two separate arguments"
+            exit 1
+        fi
+        if [[ "$vm_arg" == "cp="* ]]; then
+            custom_cp=${vm_arg#cp=}
+        elif [[ "$vm_arg" == "classpath="* ]]; then
+            custom_cp=${vm_arg#classpath=}
+        fi
+        if [[ -z "${custom_cp+x}" ]]; then
+            jvm_args+=("-${vm_arg}")
+        else
+            IFS=: read -ra custom_cp_a <<< "${custom_cp}"
+            for e in "${custom_cp_a[@]}"; do
+                absolute_cp+=("${e}")
+            done
+        fi
+    else
+        launcher_args+=("$o")
+    fi
+done
+
+cp="$(IFS=: ; echo "${absolute_cp[*]}")"
+module_path="$(IFS=: ; echo "${absolute_module_path[*]}")"
+upgrade_module_path="$(IFS=: ; echo "${absolute_upgrade_module_path[*]}")"
+
+echo "ump: ${upgrade_module_path}"
+
+if [[ "${VERBOSE_GRAALVM_LAUNCHERS}" == "true" ]]; then
+    set -x
+fi
+
+echo exec "${location}/<jre_bin>/java" "${jvm_args[@]}" \
```

Comment:
I guess this is left-over debug.
Note that it should also be possible debug this by setting the `VERBOSE_GRAALVM_LAUNCHERS` environment variable to `true`.